pypeman/channels.py

```diff
@@ -377,95 +383,218 @@ def handle_and_wait(self, msg):
+                self.logger.info("%s end handle %s", str(self), str(msg))
+
+    async def inject(self, msg, start_nodename):
+        """
+        Inject a message at a given node name
+
+        Args:
+            msg (message.Message): Message to inject
+            start_nodename (str): Node name where inject the message
+        """
+        if not start_nodename or start_nodename == "_initial":
+            result = await self._call_base_handling(
+                msg=msg, start_nodename=start_nodename)
+            return result
+
+        start_node = self.get_node(name=start_nodename)
+        if not start_node:
+            raise ValueError("Node %s not found", start_nodename)
+        logger.debug("Will inject msg %r in node %r", msg, start_node)
+        if start_node in self.init_nodes:
+            msg = await self._call_init_nodes(msg=msg, start_nodename=start_nodename)
+            await self._call_base_handling(msg=msg, start_nodename="_initial")
+        elif start_node in self._nodes:
+            await self._call_base_handling(msg=msg, start_nodename=start_nodename)
+        elif start_node in self.join_nodes:
+            await self._call_join_nodes(msg=msg, start_nodename=start_nodename)
+            base_msg_data = await self.message_store.get(msg.store_id)
+            base_msg = base_msg_data["message"]
+            await self._call_final_nodes(msg=base_msg)
+        elif start_node in self.drop_nodes:
+            await self._call_drop_nodes(msg=msg, start_nodename=start_nodename)
+            base_msg_data = await self.message_store.get(msg.store_id)
+            base_msg = base_msg_data["message"]
+            await self._call_final_nodes(msg=base_msg)
+        elif start_node in self.reject_nodes:
+            await self._call_reject_nodes(msg=msg, start_nodename=start_nodename)
+            base_msg_data = await self.message_store.get(msg.store_id)
+            base_msg = base_msg_data["message"]
+            await self._call_final_nodes(msg=base_msg)
+        elif start_node in self.fail_nodes:
+            await self._call_fail_nodes(msg=msg, start_nodename=start_nodename)
+            base_msg_data = await self.message_store.get(msg.store_id)
+            base_msg = base_msg_data["message"]
+            await self._call_final_nodes(msg=base_msg)
+        elif start_node in self.final_nodes:
+            await self._call_final_nodes(msg=msg, start_nodename=start_nodename)
+        else:
+            raise Exception("Node %r found but not in init/handle/end nodes, weird..", start_node)
+
     async def handle(self, msg):
         """ Overload this method only if you know what you are doing but
         call it from child class to add behaviour.
 
         :param msg: To be processed msg.
+        :param start_nodename: the nodename where you want to inject the message, if no nodename
+            is passed, process it from the channel's start (dont work with init nodes)
```

Comment:
dont -> doesn't

Reply:
mmmm, Iremove this part of the docstring as "start_nodename" was no more an arg of the function, and in _call_base_handling that use it, the doc is:
            start_nodename (str, optional): the nodename where you want to
                inject the message, if no nodename is passed, process it from
                the channel's start. You could set it to "_initial", the message will be processed
                from the start but will bypass init_nodes